miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java

```diff
@@ -164,19 +173,18 @@ public ModelAndView newUnassignedRun(@PathVariable String platformTypeName, Mode
   public ModelAndView setupForm(@PathVariable Long runId, ModelMap model) throws IOException {
     Run run = runService.get(runId);
 
-    return setupForm(run, run.getSequencerReference().getPlatform().getPlatformType(), model);
+    return setupForm(run, model);
 
   }
 
   @RequestMapping(value = "/alias/{runAlias}", method = RequestMethod.GET)
   public ModelAndView setupForm(@PathVariable String runAlias, ModelMap model) throws IOException {
     Run run = runService.getRunByAlias(runAlias);
-    return setupForm(run, run.getSequencerReference().getPlatform().getPlatformType(), model);
+    return setupForm(run, model);
 
   }
 
-  public ModelAndView setupForm(Run run, PlatformType platformType, ModelMap model) throws IOException {
-    model.put("platformType", platformType);
+  public ModelAndView setupForm(Run run, ModelMap model) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)